File: static/js/countdown.js
// Countdown Timer

function updateCountdown() {
    // Calculate the time remaining until Christmas
    let now = new Date();
    let christmas = new Date(now.getFullYear(), 11, 25); // December is 11 (0-based index)

    // Check if Christmas has already passed this year
    if (now > christmas) {
        christmas.setFullYear(now.getFullYear() + 1);
    }

    let timeUntilChristmas = christmas - now;

    // Convert the time remaining to days, hours, minutes, and seconds
    let days = Math.floor(timeUntilChristmas / (1000 * 60 * 60 * 24));
    let hours = Math.floor((timeUntilChristmas % (1000 * 60 * 60 * 24)) / (1000 * 60 * 60));
    let minutes = Math.floor((timeUntilChristmas % (1000 * 60 * 60)) / (1000 * 60));
    let seconds = Math.floor((timeUntilChristmas % (1000 * 60)) / 1000);

    // Update the countdown element with the calculated time
    document.getElementById("countdown").innerHTML = days + "d " + hours + "h " + minutes + "m " + seconds + "s";
}

// Update the countdown every second
setInterval(updateCountdown, 1000);

updateCountdown();

File: countdown.py
import datetime


def christmas_countdown():
    # Get the current date and time
    now = datetime.datetime.now()

    # Calculate the date for Christmas this year
    christmas = datetime.datetime(now.year, 12, 25)

    # Check if Christmas has already passed this year
    if now > christmas:
        christmas = christmas.replace(year=now.year + 1)

    # Calculate the time remaining until Christmas
    time_until_christmas = christmas - now

    # Extract days, hours, minutes, and seconds
    days = time_until_christmas.days
    seconds = time_until_christmas.seconds
    hours, seconds = divmod(seconds, 3600)
    minutes, seconds = divmod(seconds, 60)

    # Create a dictionary to store the countdown values
    countdown_values = {
        'days': days,
        'hours': hours,
        'minutes': minutes,
        'seconds': seconds
    }

    return countdown_values


if __name__ == "__main__":
    christmas_countdown()
